dotCMS/src/main/java/com/dotmarketing/business/PermissionCacheImpl.java

```diff
@@ -149,7 +151,18 @@ private Optional<String> shortLivedKey(@NotNull final Permissionable permissiona
 			return Optional.empty();
 		}
 
-		return Optional.of(permissionable.getPermissionId() + permissionType + userIn.getUserId() + respectFrontendRoles + (
+		String permissionableKey =  permissionable.getPermissionId();
+		if(permissionable instanceof Contentlet && !(permissionable instanceof Host)){
+			//We need a bit more dispersion for contentlets since they rely only on the identifier
+			//There can be cases on which the same contentlet is being checked for different permissions like for live and working content
+			//We need to be able to tell these cases apart
+			//Change directly the contentlet permissionable-id on the contentlet implementation seems to be quite risky
+			//Changing only the cache key here is safer see https://github.com/dotCMS/core/issues/30991
+		    final Contentlet contentlet = (Contentlet)permissionable;
+		    permissionableKey = contentlet.getPermissionId() + StringPool.COLON + contentlet.getInode();
```

Comment:
I thought we were going to use language id instead

Reply:
I never said that. 
Though Language does not help me to tell live and working apart